### PR TITLE
Converting to exec to properly return errors from the ci-operator

### PR DESCRIPTION
**Pull request by bradmwilliams**

### CODE REVIEW

File: prow.go [hunk 1]
```diff
@@ -369,9 +369,9 @@ func (m *jobManager) newJob(job *Job) error {
 			envPrefix := strings.Join(restoreImageVariableScript, " ")
 			container.Command = []string{"/bin/bash", "-c"}
 			if job.Mode == "build" {
-				container.Command = append(container.Command, fmt.Sprintf("registry_host=%s\n%s\n\n%s\n%s ci-operator $@ & wait", registryHost, script, permissionsScript, envPrefix), "")
+				container.Command = append(container.Command, fmt.Sprintf("registry_host=%s\n%s\n\n%s\nexec %s ci-operator $@", registryHost, script, permissionsScript, envPrefix), "")
 			} else {
-				container.Command = append(container.Command, fmt.Sprintf("registry_host=%s\n%s\n%s ci-operator $@ & wait", registryHost, script, envPrefix), "")
+				container.Command = append(container.Command, fmt.Sprintf("registry_host=%s\n%s\nexec %s ci-operator $@", registryHost, script, envPrefix), "")
```
Comment (smarterclayton) [inline prow.go:374]: are you sure this is the correct spot for exec?  The `%s` has env vars, is that correct?

File: prow.go [hunk 2]
```diff
@@ -369,9 +369,9 @@ func (m *jobManager) newJob(job *Job) error {
 			envPrefix := strings.Join(restoreImageVariableScript, " ")
 			container.Command = []string{"/bin/bash", "-c"}
 			if job.Mode == "build" {
-				container.Command = append(container.Command, fmt.Sprintf("registry_host=%s\n%s\n\n%s\n%s ci-operator $@ & wait", registryHost, script, permissionsScript, envPrefix), "")
```
Comment (smarterclayton) [inline prow.go:372]: Change the trap to have;

```
trap 'e=$!; ...; exit $e' TERM EXIT
```

File: prow.go (hunk 1 above)
Comment (bradmwilliams) [reply prow.go]: Nope, the exec fails when an environment is passed.  
Fixed.

File: prow.go (hunk 2 above)
Comment (bradmwilliams) [reply prow.go:372]: Done.

**Comment by smarterclayton** [comment]
Squash please

**Comment by bradmwilliams** [comment]
> Squash please
Done

**Pull request merged**